### DISCUSSION TITLE
Display the off road sections on route and diagram

### DISCUSSION THREAD

**Discussion by WolfgangZ42** [devemux86/cruiser#664] [Q&A] [answered]
During the ACT Portugal tour, the question was asked several times: “How much longer will we be in the dirt - when will the asphalt return?” Unfortunately, it is not possible to answer this question, neither in navigation mode nor in planning mode.

Now that we are able to display individual route segments in different colors in planning mode:
<img width="500" src="https://github.com/user-attachments/assets/71bb6cc3-d696-463c-bbfb-7bd518efcc27"/> . . . <img width="155" src="https://github.com/user-attachments/assets/cb8dcbc9-7d85-40df-93b1-10e05b898f41"/>

Would it be possible during navigation (and only there) to make the color of the route dependent on the condition defined in OSM? For example like this:
<img width="500" src="https://github.com/user-attachments/assets/a2de7fff-f883-4dcf-9e02-9aac106e2f0f"/>

In my opinion, it is sufficient to distinguish only between asphalt and gravel.

**Comment by devemux86** [comment]
@WolfgangZ42 Thanks for the suggestion, this is an interesting idea.

**Comment by devemux86** [comment]
Replying to a comment by @0709wiwiwi in https://github.com/devemux86/cruiser/discussions/250#discussioncomment-14084099:

Description may be useful in the future for waypoint descriptions in https://github.com/devemux86/cruiser/discussions/401.
So it will probably be better to store it in an extension, like the max speed.

This should work as the other route details, like max speed and street name.
Split in sections with start and finish with "--".

```xml
<max_speed>30</max_speed>
<route_information>unpaved</route_information>
...
<max_speed>--</max_speed>
<route_information>--</route_information>
```

**Comment by devemux86** [comment]
To be compatible with other routers, we only need to declare the "unpaved" sections.

As the surface tag is not always filled, perhaps it is better to process the highway tag?
(like in the map themes)

OpenRouteService documentation has a nice classification for these tags:
https://giscience.github.io/openrouteservice/api-reference/endpoints/directions/extra-info/surface
https://giscience.github.io/openrouteservice/api-reference/endpoints/directions/extra-info/waytype

**Reply by FishEy3** [reply]
@devemux86 
It's true that the surface tag is often left blank.
The **highway=track** tag should indicate unpaved roads (I specify "should" because some regions allow exceptions).

In some cases, I've driven on unpaved roads mapped with the **highway=track** and **tracktype=grade(1/5)** tags together, hoping they'll be helpful.

I'll provide some examples of unpaved routes mapped in different ways.
<img width="398" height="196" alt="image" src="https://github.com/user-attachments/assets/772da884-e531-482a-9e38-4b0613adfcd4" />
<img width="410" height="137" alt="image" src="https://github.com/user-attachments/assets/fe7ea6f5-6830-4640-8124-0a346ad5aea4" />
<img width="407" height="271" alt="image" src="https://github.com/user-attachments/assets/c7495c9f-7f5b-4b76-870b-9a51294bee9d" />

**Reply by devemux86** [reply]
"Unpaved" can be defined after examining tags, for example:
-  `surface` **not** (`asphalt` or `concrete` or `paved` or ...)

- Or `highway` =  `path` or `track` or ...

---

The file formats (json, gpx, ...) do not need to store all OSM tags, as they are not useful.
Usually we only need to know if a road is "unpaved" or not. Not its exact surface details.

**Reply by devemux86** [reply]
So we just store the final result, which route sections are "unpaved".

Remember routers with avoid options have a generic "Avoid unpaved".
What is "unpaved" is processed internally by the routers and the apps.

**Reply by 0709wiwiwi** [reply]
Just one single note.  Know  I've seen more users that requested and prefer to cycle (or not) onto **gravel** roads. They're usually the types of users  who do prefer cycling through forests roads (if still existant) instead of the crazy hustle and bustle of the regular roads. So, everyone has just simply their own preference.

**Reply by devemux86** [reply]
[Gravel](https://wiki.openstreetmap.org/wiki/Tag:surface%3Dgravel) is considered in the "unpaved" category.
It can be used in Cruiser when enabling the off-road tracks in the settings.

[Above](https://github.com/devemux86/cruiser/discussions/664#discussioncomment-14091601) is a current proposal for what can be considered as "unpaved": everything that is not in "paved" category.

**Reply by 0709wiwiwi** [reply]
> What is “unpaved” is processed internally by the routers and the apps.

It would be highly unusual for all (6) routers to indicate this uniformly. 
so I suspect that the Cruiser app will ultimately take care of this itself?

**Reply by devemux86** [reply]
@0709wiwiwi That's right, each router supports it (or not) differently.

**Comment by devemux86** [comment]
We can color the off road sections on the route and diagram.

Also display the **percentage** of the off road sections.

- **Enable**: Settings | Routing | Appearance | Off road tracks

- **Select color**: Settings | Routing | Appearance | Colors | Off road tracks

<img src="https://github.com/user-attachments/assets/66ab1d16-55d8-4a43-b355-29a10a30eb95" width="50%"/>

**Reply by FishEy3** [reply]
@devemux86 
> Currently we examine the [surface](https://wiki.openstreetmap.org/wiki/Key:surface) tag and consider as unpaved sections that:
> 
> * Are not tagged with surface as: `asphalt` or `concrete` or `paved`
> * Or do not have a surface tag (is this correct ?)

I'll do some testing; often, the lack of the SURFACE tag does NOT indicate an UNPAVED section.
The person who mapped on OSM might simply not have inserted that tag.
I would consider UNPAVED where it is specified in SURFACE or in TRACKTYPE 2 to 5 (1 indicates a PAVED road).

**Reply by devemux86** [reply]
@FishEy3 

> often, the lack of the SURFACE tag does NOT indicate an UNPAVED section.
The person who mapped on OSM might simply not have inserted that tag.

We can change this, if needed.

> I would consider UNPAVED where it is specified in SURFACE or in TRACKTYPE 2 to 5 (1 indicates a PAVED road).

We can base the process on one tag each time, either `surface` or `highway`.
Please see [above](https://github.com/devemux86/cruiser/discussions/664#discussioncomment-14091601) for an older suggestion.

**Reply by devemux86** [reply]
Road tags are easy to check on BRouter website with its Data table tool, see the instructions here:

- https://github.com/devemux86/cruiser/discussions/272

**Reply by devemux86** [reply]
The next update will examine multiple tags simultaneously for better results:
`surface`, `highway`, `tracktype`, etc.

**Reply by WolfgangZ42** [reply]
From July 1st I stayed with 5.4.2 because of the tiles - but now I'm back on the route (navigation jargon). Nevertheless: I miss the tiles and in my eyes the usability has suffered. But I'm getting used to the new approach.
And I get the new functions in return - I wouldn't want to miss this one in particular (#664)!
Just perfect at the weekend on the motorcycle:
<img height="200" src="https://github.com/user-attachments/assets/1493800f-a8de-4132-bd44-082bc27aab9d"/>
Only very few routes, which were probably only recently paved, were not yet properly defined in OSM.

However, I have a small problem with my bicycle tours. All cycle routes are shown as off road, even though they are paved. Here is an example:
<img height="500" src="https://github.com/user-attachments/assets/a7243415-c084-4db7-84a0-811e990650a0"/> . . . <img height="500" src="https://github.com/user-attachments/assets/d60091f1-8286-4336-862f-c93ba8870977"/>

I think it is due to highway=path. Maybe surface=asphalt can be given a higher priority. I would like to show my fellow riders an exact percentage of gravel: 
<img height="190" src="https://github.com/user-attachments/assets/7da7e879-5ec7-48fc-9edb-13b1b79b87db"/>

Thank you for considering.

**Reply by devemux86** [reply]
@WolfgangZ42 Thanks for the report, it will be improved in the next update.

**Reply by FishEy3** [reply]
One question: I use Mapsforge offline maps. Is the data taken only from these maps (if configured)?
I ask because updates arrive at variable intervals (for example, Mapforge Italy maps are updated sometimes monthly, sometimes after 3 months) and we may get results that differ from those available on OSM. In this case, it wouldn't depend on Cruiser.

**Reply by devemux86** [reply]
@FishEy3

Mapsforge offline maps are created worldwide for all countries.
And they are only used for map rendering (not routing).

Map themes have the layer "Off road tracks", if you want to display it.

**Offline routing uses its own data which is updated daily.**
(menu Routing | BRouter)

**Reply by FishEy3** [reply]
@devemux86 
Thanks for the explanation, I have been using the off road layer for a long time, it is useful

**Reply by WolfgangZ42** [reply]
Can the "paving stones" surface also be considered a non-offroad, please:
![2025-09-26 07_14_23-iD – Radweg _ OpenStreetMap](https://github.com/user-attachments/assets/f3a69b4c-306d-43d2-971b-76bd15a7eb38)

A large number of cycle paths are equipped in this way. Thank you!

**Reply by devemux86** [reply]
@WolfgangZ42 Currently, we consider the surface to be unpaved if it is not (asphalt, concrete, paved).

We can always improve the detection using more surface tags.

**Reply by devemux86** [reply]
@WolfgangZ42 We better use more paved tags as described in documentation:

- https://wiki.openstreetmap.org/wiki/Key:surface#Paved

- https://giscience.github.io/openrouteservice/api-reference/endpoints/directions/extra-info/surface

**Reply by devemux86** [reply]
@WolfgangZ42 Also it's useful to examine how the BRouter profiles define the `ispaved`:

- https://github.com/abrensch/brouter/blob/master/misc/profiles2/trekking.brf#L80

**Reply by WolfgangZ42** [reply]
Sure we should take the same consideration as BRouter - ispaved seems to be perfect:

        > assign ispaved = surface=paved|asphalt|concrete|paving_stones|sett

Openrouteservice even includes more detailed surfaces into "paving stones":

<img width="768" height="75" alt="image" src="https://github.com/user-attachments/assets/2a730d17-8c2a-4582-a05c-b077d8b547f1" />

Thanks again for implementing the different colors on route and diagram!

**Reply by devemux86** [reply]
@WolfgangZ42 We can include all values mentioned by OpenRouteService in the categories:
paved, asphalt, concrete and paving stones.

**Comment by devemux86** [comment, marked as answer]
- Implemented in https://github.com/devemux86/cruiser/discussions/872

**Comment by WolfgangZ42** [comment]
It's me again ... 🥴

When I recently showed some friends how to display off-road routes in Cruiser, we unfortunately encountered a problem. 

We are planning to drive the ACT Montenegro+Albania and noticed that the percentage of gravel parts is estimated much lower by Cruiser than specified by ACT. A few random checks revealed that long stretches in Albania were defined as _highway=unclassified_ without specifying _surface=*_, which Cruiser interprets as **paved**. The latter may be correct for some countries, but probably not for most - see wiki.openstreetmap.org/wiki](https://wiki.openstreetmap.org/wiki/Tag:highway%3Dunclassified)

In the future, could Cruiser please interpret _highway=unclassified without surface=*_ as **unpaved**? I would really appreciate that. Thanks for your consideration.

**Reply by WolfgangZ42** [reply]
An example: [in OSM](https://www.openstreetmap.org/edit#map=17/41.835142/20.008418) 
Before this point surface=ground was specified and after it surface=* is missing - resulting in this:
![20251121_165206492](https://github.com/user-attachments/assets/188e5f2e-ec76-452b-a45e-338514cfa557)

**Reply by MotoUKRider** [reply]
@WolfgangZ42 This could have a knock-on effect for others, so I strongly suggest that if this is implemented, it is an _option_ in the Routing Profile.  I have plenty of `unclassified` roads in my rural part of the UK which are normal tarmac albeit single-track and very much like the picture in the Wiki article linked above.

This would also be a similar "assumption" to the common occurrence of missing speed-limit tags: What would we assume in that scenario as not all road types have the same limits...

**Reply by WolfgangZ42** [reply]
Agreed! It is an assumption - no question. 
The best way of course would be to correct the OSM definition - lot of work I guess.
If it could implemented as an option, I would be happy. But that's of course a complication in setup the routing option ...

**Reply by devemux86** [reply]
@WolfgangZ42 Do you calculate the route with BRouter (offline)?

There are 2 separate things:

- **Route appearance**: we can improve what is displayed as "off road" in route and diagram.

- **Route calculation**: to change the calculated route vs "off road", you need to improve the OSM tags.

> The best way of course would be to correct the OSM definition

:+1:

**Reply by devemux86** [reply]
In general, changing the way a routing service or application works,
because OSM tags are incorrect or missing, does not seem the right process.

If the "surface" tag is missing, is the road considered unpaved or not?

Isn't the right proces to add "surface" tags or fix them when there are off-road sections?

Otherwise, making assumptions can have strange results.

**Reply by devemux86** [reply]
Or we could display differently the sections with unknown (i.e. missing) surface,
although this can get complicated.

**Reply by MotoUKRider** [reply]
Of the many `unclassified` roads I looked at here, only 2 had a surface tag (asphalt). Convention seems to be that a "road" has a smooth surface unless it is tagged otherwise. I would have expected anything "lumpy" and/or "loose" to be a `track` ? (the application of `track` in this instance is encouraged in the Wiki "if not fit for a car, but OK for 4x4, 2 wheels/walking").

Example of an `unclassified` road junction: Surface is standard although it's been fixed a few times and not generally flat. No speed limit in the tags, but I know it's a _national speed limit_ (60mph). Usual caution expected: Cyclists, walkers, horses, farm traffic and surface "issues".
![cruiser-unclassified-example](https://github.com/user-attachments/assets/0d5a3c04-6306-491f-b6a8-b94d1b1663dd)

I wonder how ACT have calculated their surface ratio? I am guessing by riding it. Is it reasonable to assume that those who seek out "special" parts of a map should/would be more adept at housekeeping OSM rather than rely on an App to supply correct data?

**Reply by WolfgangZ42** [reply]
> @MotoUKRider: I am guessing by riding it.

That seems to be the case to me too - unfortunately without housekeeping! 

> @devemux86: There are 2 separate things:
**Route appearance:** we can improve what is displayed as "off road" in route and diagram.
**Route calculation:** to change the calculated route vs "off road", you need to improve the OSM tags.

I must admit that I only thought about the route display, but of course this may look completely different for other users. 

> @devemux86: could display differently the sections with unknown (i.e. missing) surface

Perhaps a solution. Although "paved," "unpaved," and "unknown" complicate the situation considerably. 
Couldn't there be a switch under `Settings/Routing/Advanced` that assigns "unknown" to either the "paved" or "unpaved" category?

**Reply by WolfgangZ42** [reply]
> @MotoUKRider: should/would be more adept at housekeeping OSM

As I said before, that's the best way. I did it already for the first day of the ACT MTN-AL - really time-consuming ... and just related to satellite maps.
It should be standard practice for ACT to ensure that the routes distributed are properly defined in OSM. I will suggest this to them.

**Reply by WolfgangZ42** [reply]
Btw: in BRouter the different state of the Route mentioned [above](https://github.com/devemux86/cruiser/discussions/664#discussioncomment-15039815) is shown:
![2025-11-22 13_25_52-BRouter Web Client und 1 weitere Seite - Persönlich – Microsoft​ Edge](https://github.com/user-attachments/assets/09657659-032b-42b6-8fda-86de732392c3)
https://brouter.de/brouter-web/#map=17/41.83467/20.01082/osm-mapnik-german_style

**Reply by devemux86** [reply]
@WolfgangZ42 

This is map rendering via the render theme and it displays the unpaved "surface=ground" differently
from the other paved road or with missing surface that is displayed with the normal white fill color.

So the missing surface is displayed with normal white fill color, it does not have a special rendering.

**Reply by devemux86** [reply]
@WolfgangZ42 The same rendering also exists in OpenStreetMap website.
https://www.openstreetmap.org/way/662178229
https://www.openstreetmap.org/way/662178174
